server/slack_notifier.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/applikatoni/applikatoni/deploy"
+	"github.com/applikatoni/applikatoni/models"
+	"log"
+	"net/http"
+)
+
+type slackMsg struct {
+	Text      string `json:"text"`
+	Username  string `json:"username"`
+	IconEmoji string `json:"icon_emoji"`
+}
+
+func NotifySlack(db *sql.DB, deploymentId int) {
+	deployment, err := getDeployment(db, deploymentId)
+	if err != nil {
+		log.Printf("Could not find deployment with id %v, %s\n", deploymentId, err)
+		return
+	}
+
+	application, err := findApplication(deployment.ApplicationName)
+	if err != nil {
+		log.Printf("Could not find application with name %v, %s\n", deployment.ApplicationName, err)
+		return
+	}
+
+	target, err := findTarget(application, deployment.TargetName)
+	if err != nil {
+		log.Printf("Could not find target with name %v, %s\n", deployment.TargetName, err)
+		return
+	}
+
+	if target.SlackUrl == "" {
+		return
+	}
+
+	SendSlackRequest(deployment, target, application)
+}
+
+func SendSlackRequest(d *models.Deployment, t *models.Target, a *models.Application) {
+	payload, err := json.Marshal(slackMsg{
+		IconEmoji: ":it:",
+		Username:  "Applikatoni Bot",
+		Text: fmt.Sprintf("Successfully deployed %s :tada:\n to: %s \n on: %s %s",
+			d.ApplicationName,
+			d.TargetName,
+			d.Branch,
+			d.CommitSha),
+	})
+
+	if err != nil {
+		log.Printf("Error marshaling Slack notification\n")
```

Comment:
ah, can I assume that there is a similarly missing return in the bugsnag notifier test?
https://github.com/applikatoni/applikatoni/blob/master/server/bugsnag_notifier_test.go#L40

Also, I noticed that bug snag and my new test file output to the logs when running tests. is that expected?